### DISCUSSION TITLE
Can `DatabaseErrorInformation` contain db values?

### DISCUSSION THREAD

**Discussion by agostbiro** [diesel-rs/diesel#3289] [Q&A] [answered]
Hi,

I'm wondering if it's possible that the [`DatabaseErrorInformation`](https://docs.diesel.rs/master/diesel/result/trait.DatabaseErrorInformation.html#tymethod.message) contains database values? I'm asking, because I would like to log its contents.

Thanks,
Agost

**Comment by weiznich** [comment, marked as answer]
We cannot grantee that any type implementing `DatabaseErrorInformation` will not contain database values. This trait is implementation specific, which means that any potential third party implementation can provide whatever information they want. For the implementations provided in diesel this mostly closely mirrors whatever the underlying database library provides as information about error messages. Again we do not have any control about the exact error messages emitted there, we just forward them to users. In the end it highly depends on your database system, your database connection library version (so the version of libpq, libmysqlclient or libsqlite3) and what you would expect to be critical to log or not. Depending on the database system and the error case these trait implementations might contain parts of the executed query.

**Reply by agostbiro** [reply]
Thanks, appreciate the quick reply. In my case, I'm using Diesel for a local E2EE app with SQlite, and the error logs would be sent to Sentry, so best not to include the `DatabaseErrorInformation`.
